docs/setup-project.adoc

```diff
@@ -0,0 +1,47 @@
+
+Hercules CI is configured "by convention" using `.nix` files in your repository.
+Follow these steps to ensure a correct setup.
+
+// TODO: If you haven't set up a project with Nix before, we recommend the <<tutorial>>
+
+1. Clone your repository
+
+2. Choose which Nix file to use for CI
++
+Reuse `/default.nix`::
+For simple projects that only need to build for a single architecture and a single set of dependencies.
+or create a `/ci.nix` or `/nix/ci.nix`::
+For projects that need to build for multiple architectures or versions, or where `default.nix` already
+has a special meaning that conflicts with the need for CI.
+
+3. Use pinned dependencies instead of `<nixpkgs>` or similar.
++
+Hercules CI uses an empty `NIX_PATH` to help with evaluation reprodicibility.
++
+The unofficial NixOS wiki has https://nixos.wiki/wiki/FAQ/Pinning_Nixpkgs[a basic method of pinning Nixpkgs].
++
+You can use an impure evaluation-time fetcher such as `builtins.fetchUrl`, although this typically makes your evaluation unreproducible.
+
+4. Test locally with `nix-instantiate`
++
+The behavior of `nix-instantiate` serves as a reference for evaluation in Hercules CI.
+You may use the following command to test locally:
++
+[source,bash]
+----
+NIX_PATH="" nix-instantiate nix/ci.nix | tee /dev/stderr | wc -l
+----
++
+WARNING: Due to import from derivation, seeing any output is not sufficient. Check that the last line of output is `1` or more. This is the number of derivations to build from traversing the Nix expression.
```

Comment:
```suggestion
Make sure to wait for the process to exit and inspect all derivations are present in the output.
```

Reply:
Apparently output can be deceiving, which is why I added the `wc -l`. I want these instructions to be watertight. (also here: Maybe quickstart should be the goal of this guide and nixifying the subject of another guide, in which case this text will be moved to the other guide)

Reply:
If feels quite magical to me, I had hard time understanding why it's going through the hoops.

Maybe it's better to say `nix-instantiate 2>/dev/null` for the list of derivations? Although they will get the same list on CI anyway so the first starting goal should be for `nix-instantiate` to just succeed.